src/userauth.c

```diff
@@ -1299,6 +1308,21 @@ size_t plain_method(char *method, size_t method_len)
     return method_len;
 }
 
+/* Function to check if the given version is less than pattern (OpenSSH 7.8)
+ * Returns 1 if the version is less than OpenSSH_7.8, 0 otherwise
+ */
+static int is_version_less_than_78(const char *version)
+{
+    int major, minor;
+    if(sscanf(version, "OpenSSH_%d.%d", &major, &minor) == 2) {
```

Comment:
`sccanf()` is considered slow and heavy, and it would be its only use in libssh2.
Would there be a way to replace it with e.g. `strtol()`? (we already use this one)

Reply:
@vszakats Removed sscanf. Made changes to just extract the required version char without using any library functions.